covalent_ui/webapp/src/App.js

```diff
@@ -20,69 +20,88 @@
  * Relief from the License may be granted by purchasing a commercial license.
  */
 
-import { useEffect } from 'react'
-import { useDispatch } from 'react-redux'
-import CssBaseline from '@mui/material/CssBaseline'
-import ThemeProvider from '@mui/system/ThemeProvider'
-import { Helmet, HelmetProvider } from 'react-helmet-async'
-import { Routes, Route } from 'react-router-dom'
+ import { useEffect } from 'react'
+ import { useDispatch } from 'react-redux'
+ import CssBaseline from '@mui/material/CssBaseline'
+ import ThemeProvider from '@mui/system/ThemeProvider'
+ import { Helmet, HelmetProvider } from 'react-helmet-async'
+ import { Routes, Route } from 'react-router-dom'
 
-import './App.css'
-import Dashboard from './components/Dashboard'
-import socket from './utils/socket'
-import { fetchResult } from './redux/resultsSlice'
-import { setLattice } from './redux/latticePreviewSlice'
-import theme from './utils/theme'
-import { ReactFlowProvider } from 'react-flow-renderer'
-import LatticePreviewLayout from './components/preview/LatticePreviewLayout'
-import DispatchLayout from './components/dispatch/DispatchLayout'
-import NotFound from './components/NotFound'
+ import './App.css'
+ import Dashboard from './components/Dashboard'
+ import socket from './utils/socket'
+ import { fetchResult } from './redux/resultsSlice'
+ import { setLattice } from './redux/latticePreviewSlice'
+ import theme from './utils/theme'
+ import { ReactFlowProvider } from 'react-flow-renderer'
+ import LatticePreviewLayout from './components/preview/LatticePreviewLayout'
+ import DispatchLayout from './components/dispatch/DispatchLayout'
+ import NotFound from './components/NotFound'
+ import { differenceInSeconds} from 'date-fns'
 
-const App = () => {
-  const dispatch = useDispatch()
+ const App = () => {
+   const dispatch = useDispatch()
 
-  useEffect(() => {
-    const onUpdate = (update) => {
-      dispatch(
-        fetchResult({
-          dispatchId: update.result.dispatch_id,
-          resultsDir: update.result.results_dir,
-        })
-      )
-    }
+   useEffect(() => {
```

Comment:
This implementation seems to be effectively re-implementing functionality similar to debouncing which is already included in the lodash library (and tested). All we would need to debounce is the on result update callback. If this can be done in a future PR that would be ideal.